client/scala/src/main/scala/ai/verta/repository/WalkOutput.scala

```diff
@@ -0,0 +1,14 @@
+package ai.verta.repository
+
+/** A class to represent the output of Commit's walk method
+ *  @param folderPath path to current folder
+ *  @param folderNames names of subfolders in current folder
+ *  @param blobNames names of blobs in current folder
+ *  @param remainingLocations list of remaining locations
+ */
+class WalkOutput(
```

Comment:
What do you expect the user to do with this class? I'd recommend adding some methods to make it simpler for the user to know what to do.

Reply:
Hmm I'm not quite sure what the user can do other than accessing with the fields of WalkOutput. This is just a fancy version of Tuple4, making it more semantically meaningful.

Reply:
Essentially, imagine the user is typing in their IDE and presses `.` in this object. This should list values that they can access and common actions they can make. For example, `Try.get` can be implemented via a `match`, but that's very verbose. So the action of getting the value is provided. Similarly for `toOption` and many others.

When you define a class, I recommend thinking about what the user will do with the class and give them a method for the most common cases, so that they don't have to see the objects available, interpret what they mean and their types, then thing which actions are available on those types.